### PR TITLE
Fix: DBInterface::get() return nullable strings

### PR DESCRIPTION
**Pull request by qiluo-msft**

### CODE REVIEW

**Review by arlakshm — APPROVED** [review]

File: common/dbinterface.cpp [hunk 1]
```diff
@@ -67,9 +67,9 @@ std::string DBInterface::get(const std::string& dbName, const std::string& hash,
             throw UnavailableDataError(message, hash);
         }
         const std::string& value = *pvalue;
-        return value == "None" ? "" : value;
+        return value == "None" ? shared_ptr<string>() : make_shared<string>(value);
     };
-    return blockable<std::string>(innerfunc, dbName, blocking);
+    return blockable<shared_ptr<string>>(innerfunc, dbName, blocking);
```
Comment (kcudnik) [inline common/dbinterface.cpp:72]: inconsistent namespace usage, std::shared_ptr in function return value vs shared_ptr in function body

Comment (kcudnik) [reply common/dbinterface.cpp:72]: same for std::string and string

Comment (qiluo-msft) [reply common/dbinterface.cpp]: Fixed

**Review by kcudnik — APPROVED** [review]

**Pull request merged**